routes/index.js

```diff
@@ -1,55 +1,473 @@
-var express = require('express');
-var router = express.Router();
-var Song = require('../models/song');
-var User = require('../models/user')
-var app = require('../app')
-var pdf = require('html-pdf')
-var fs = require('file-system')
-
-
-/* GET home page. */
-router.get('/', function(req, res, next) {
-    Song.find(function(err, songs) {
-            Song.find({
-                source: {
-                    $ne: undefined
-                }
-            }, function(err, demo) {
-                console.log(demo)
+var express = require('express'),
+    router = express.Router(),
+    Song = require('../models/song'),
+    User = require('../models/user'),
+    _ = require('lodash'),
+    Playlist = require('../models/playlist'),
+    async = require('async')
+
+
+router.get('/temp', function(req, res){
+  for (var i = 0; i < 20; i++){
+    new Song({
+      title: 'english song ' + i,
+      lang: ''
+    }).save()
+  }
+  res.send('success')
+})
+
+router.get('/', function(req, res) {
+    console.log()
+    var messages = req.flash()
+    Song.find(function(err, songs, count) {
+            if (err) {
+                res.status(400).send('error getting song list ' + err)
+            }
+            if (req.isAuthenticated()) {
+                User.findOne({
+                    _id: req.user._id
+                }, function(err, user) {
+                    if (err) return handleError(err)
+                    Playlist.find({
+                        owner: req.user._id
+                    }, function(err, playlists) {
+                        if (err) return handleError(err)
+                        res.render('index', {
+                            songs: songs,
+                            inLibrary: user.library,
+                            playlists: playlists,
+                            messages: messages
+                        })
+                    })
+
+                })
+            } else {
+                res.render('index', {
+                    songs: songs,
+                    inLibrary: [],
+                    playlists: [],
+                    messages: messages
+                })
+            }
+        })
+        .sort({
+            title: 1
+        })
+        .limit(10)
+})
+
+router.put('/', function(req, res) {
+  var langShown = req.body.langShown.toLowerCase()
+  var langFilter = req.body.langFilter.map((lf) => lf.toLowerCase())
+  var totalSongsDisplayed = req.body.totalSongsDisplayed
+  var songs2d = []
+  async.waterfall([
```

Comment:
To help with code readability, what if you pulled out these functions and gave them a name so that I could read the waterfall?
`async.waterfall([findSongs, getTranslations, applyFilter, ... ])`